src/client/lazy-app/Compress/index.tsx

```diff
@@ -374,6 +376,22 @@ export default class Compress extends Component<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props, prevState: State): void {
+    const wasLoading =
+      prevState.loading ||
+      prevState.sides[0].loading ||
+      prevState.sides[1].loading;
+    const isLoading =
+      this.state.loading ||
+      this.state.sides[0].loading ||
+      this.state.sides[1].loading;
+    const sourceChanged = prevState.source !== this.state.source;
+    if (wasLoading !== isLoading || sourceChanged) {
```

Comment:
or: 
```suggestion
    const sourceChanged = prevState.source !== this.state.source;
    const leftImageChanged = prevState.sides[0].file !== this.state.sides[0].file;
    const rightImageChanged = prevState.sides[1].file !== this.state.sides[1].file;
    const imageChanged = sourceChanged || leftImageChanged || rightImageChanged;
    if ((wasLoading !== isLoading) || imageChanged) {
```

Reply:
Nah, what you've got here is good. When the sides change, they don't change filename.